node/http/src/api.rs

```diff
@@ -177,10 +179,16 @@ impl HttpApi {
         }
     }
 
-    pub fn submit_transaction(&self, r: SignedTransaction) -> Result<(), &str> {
+    pub fn submit_transaction(
+        &self,
+        r: &SignedTransaction,
+    ) -> Result<SubmitTransactionResponse, &str> {
         debug!(target: "near-rpc", "Received transaction {:?}", r);
-        self.submit_txn_sender.clone().try_send(r).map_err(|_| {
+        self.submit_txn_sender.clone().try_send(r.clone()).map_err(|_| {
```

Comment:
Regarding the argument `r`. How it was before is more flexible, e.g. if we have a transaction then we want the `submit_txn_sender` to take ownership of then we can just pass it to submit_transaction. If we have a transaction that we don't want to give ownership of then we clone it and then pass to `submit_transaction`, but with this change the cloning cannot be avoided and will always happen.

Reply:
seems like we can easily refactor this function if we want to avoid this clone and give ownership. this just seemed like the most straightforward way to get the hash after a move